src/frontend/src/instance/standalone.rs

```diff
@@ -85,3 +78,83 @@ impl RegionRequestHandler for StandaloneRegionRequestHandler {
             .context(HandleRequestSnafu)
     }
 }
+
+pub(crate) struct StandaloneDatanode(pub(crate) RegionServer);
+
+#[async_trait]
+impl Datanode for StandaloneDatanode {
+    async fn handle(&self, request: RegionRequest) -> MetaResult<AffectedRows> {
+        let body = request.body.context(meta_error::UnexpectedSnafu {
+            err_msg: "body not found",
+        })?;
+        let resp = self
+            .0
+            .handle(body)
+            .await
+            .map_err(BoxedError::new)
+            .context(meta_error::ExternalSnafu)?;
+
+        Ok(resp.affected_rows)
+    }
+
+    async fn handle_query(&self, request: QueryRequest) -> MetaResult<SendableRecordBatchStream> {
+        self.0
+            .handle_read(request)
+            .await
+            .map_err(BoxedError::new)
+            .context(meta_error::ExternalSnafu)
+    }
+}
+
+pub struct StandaloneDatanodeManager(pub RegionServer);
+
+#[async_trait]
+impl DatanodeManager for StandaloneDatanodeManager {
+    async fn datanode(&self, _datanode: &Peer) -> DatanodeRef {
+        Arc::new(StandaloneDatanode(self.0.clone()))
+    }
+}
+
+pub(crate) struct StandaloneTableCreator {
+    table_id_sequence: SequenceRef,
+}
+
+impl StandaloneTableCreator {
+    pub fn new(kv_backend: KvBackendRef) -> Self {
+        Self {
+            table_id_sequence: Arc::new(Sequence::new(TABLE_ID_SEQ, 1024, 10, kv_backend)),
+        }
+    }
+}
+
+#[async_trait]
+impl TableCreator for StandaloneTableCreator {
```

Comment:
Feel that the trait name is easily misleading. It only gives part of the table meta not really to create tables.